### PR TITLE
HB-04

### PR DESCRIPTION
**Pull request by Max-Pochepets**
Implemented complex relations

- Models created: CinemaHall and MovieSession
- Dao's created
- Services created
- Relations between CinemaHall and MovieSession implemented
- Methods being tested in Main

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,6 +23,11 @@
             <artifactId>hibernate-core</artifactId>
             <version>5.4.27.Final</version>
         </dependency>
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-java8</artifactId>
+            <version>5.4.27.Final</version>
+        </dependency>
```
Comment (IvanLuchkin) [inline pom.xml:30]: Do we need this?

File: src/main/java/cinema/shop/App.java [hunk 2]
```diff
@@ -1,17 +1,42 @@
...
+import cinema.shop.service.CinemaHallService;
+import cinema.shop.service.MovieService;
+import cinema.shop.service.MovieSessionService;
+import java.time.LocalDateTime;
 
 public class App {
     private static Injector injector = Injector.getInstance("cinema.shop");
 
     public static void main(String[] args) {
+
```
Comment (IvanLuchkin) [inline src/main/java/cinema/shop/App.java]: Redundant empty line

**Review by IvanLuchkin — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (Max-Pochepets) [reply pom.xml:30]: App doesn't work without it

File: src/main/java/cinema/shop/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            Query<MovieSession> getSessionsByDate
+                    = session.createQuery("select ms from MovieSession ms "
+                    + "left join fetch ms.movie "
+                    + "left join fetch ms.cinemaHall "
+                    + "where ms.movie.id =:movie_id "
+                        + "and date_format(ms.showTime, '%Y-%m-%d')=:date", MovieSession.class);
+            getSessionsByDate.setParameter("movie_id", movieId);
+            getSessionsByDate.setParameter("date", DateTimeFormatter.ISO_LOCAL_DATE.format(date));
+            return getSessionsByDate.getResultList();
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/shop/dao/impl/MovieSessionDaoImpl.java:32]: catch?

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**